src/watchmaker/__init__.py

```diff
@@ -61,14 +79,21 @@ def __init__(
                 level=logging.DEBUG)
             self.logger = logging.getLogger()
             self.logger.info('\n\n\n{0}'.format(datetime.datetime.now()))
-        elif stream:
-            self.logger.error('{0} does not exist'.format(log_path))
+        elif arguments.logger:
+            self.logger.error('{0} does not exist'.format(arguments.log_path))
         else:
             self.logger.warning('Stream logger is not enabled!')
 
         self.logger.info('Parameters:  {0}'.format(self.kwargs))
         self.logger.info('System Type: {0}'.format(self.system))
 
+    def _validate_url(self, url):
+
+        if not validators.url(url):
```

Comment:
Current:
```
import validators

def valid(url):
    if not validators.url(url):
        return False
    else:
        return True        

valid('https://www.google.com')
True

valid('https://www.google.c')
False
```

Suggested:
```
def valid(url):
    return validators.url(url)
   
valid('https://www.google.com')
True

valid('https://www.google.c')
ValidationFailure(func=url, args={'public': False, 'value': 'https://www.google.c'})
```

I think the current approach is correct so that we dont have use a try to validate.